app/controllers/submissions_controller.rb

```diff
@@ -0,0 +1,108 @@
+class SubmissionsController < ApplicationController
+
+  def new
+    @project = Project.find(params[:project_id])
+    @submission = @project.submissions.build
+    @submission.file = @project.directory
+    @submission.cores = '28'
+    @submission.extra = '-verb'
+    @submission.email = false
+    @submission.scheduled_hrs = 1
+  end
+
+  def create
+    @project = Project.find(params[:project_id])
+
+    if @submission = @project.submissions.create(submission_params)
+      redirect_to project_path(@project), notice: 'Submission successfully created.'
+    else
+      redirect_to project_path(@project), alert: "Unable to create submission: #{@submission.errors.to_a}"
+    end
+  end
+
+  def edit
+    @project = Project.find(params[:project_id])
+    @submission = @project.submissions.find(params[:id])
+  end
+
+  def update
+    @project = Project.find(params[:project_id])
+    @submission = @project.submissions.find(params[:id])
+
+    if @submission.update(submission_params)
+      redirect_to @project, notice: 'Submission updated.'
+    else
+      redirect_to @project, alert: "Unable to update submission: #{@submission.errors.to_a}"
+    end
+  end
+
+  def destroy
+    @project = Project.find(params[:project_id])
+    @submission = @project.submissions.find(params[:id])
+
+    if @submission.destroy
+      redirect_to @project, notice: 'Submission deleted.'
+    else
+      redirect_to @project, alert: "Unable to delete submission: #{@submission.errors.to_a}"
+    end
+  end
+
+  def submit
+    @project = Project.find(params[:project_id])
+    @submission = @project.submissions.find(params[:submission_id])
+    @submission.project_dir = @project.directory
+
+    if @submission.submit
+      redirect_to @project, notice: 'Job submission succeeded'
+    else
+      redirect_to @project, alert: "Job submission failed #{@submission.errors.to_a}"
+    end
+  end
+  
+  def show
+    @project = Project.find(params[:project_id])
+    @submission = Submission.find(params[:id])
+
+    respond_to do |format|
+      format.html { render :show }
+      format.json { render json: @submission }
+    end
+  end
+
+  def jobs
+    @project = Project.find(params[:project_id])
+    @submission = Submission.find(params[:submission_id])
+    @jobs = @submission.jobs.where(submission_id: params[:submission_id])
+
+    @jobs.map(&:update_status)
+
+    respond_to do |format|
+      format.json { render json: @jobs }
+    end
+  end
+
+  def index
+    @project = Project.find(params[:project_id])
+    last_jobs = []
+
+    @project.submissions.each do |sub|
+      last_job = sub.jobs.first
+      last_job&.update_status
```

Comment:
I think you meant `last_job.update_status` instead of `last_job&.update_status`?

Reply:
never mind, I didn't realize Ruby 2.3.0 let you do that! Behind the times!